client/build.rs

```diff
@@ -22,121 +15,38 @@ fn main() -> Result<()> {
     // TODO: check if this was causing the recursive loop.
```

Comment:
can you check this now?

Reply:
should you also add `println!("cargo:rerun-if-changed=build.rs");`?

Reply:
It will be redundant ([link](https://doc.rust-lang.org/cargo/reference/build-scripts.html#cargorerun-if-changedpath)).

> Cargo automatically handles whether or not the script itself needs to be recompiled, and of course the script will be re-run after it has been recompiled. Otherwise, specifying build.rs is redundant and unnecessary.

Reply:
> can you check this now?

actually, you can remove this comment now?

> It is redundant

From the doc you linked I thought that adding this will prevent build script from being run every time

Reply:
Yeah, I think, I can simple remove this commented code at all.

Intially it was done to always rebuild wasm if some sub-dependency such as `data_model` has been changed.

For now, I believe, we no longer need it (and we lived without it normally). We don't need it because right now we only do `cargo check` in `build.rs`. And that should be deprecated in #3224 when we introduce this check in CI.

After all we build permission validators in `kagami genesis`, so it will be always updated.